test/jdk/java/lang/invoke/VarHandles/VarHandleTestReflection.java

```diff
@@ -52,15 +53,33 @@ static VarHandle handle() throws Exception {
     }
 
     @Test(dataProvider = "accessModesProvider", expectedExceptions = IllegalArgumentException.class)
-    public void methodInvocation(VarHandle.AccessMode accessMode) throws Exception {
+    public void methodInvocationArgumentMismatch(VarHandle.AccessMode accessMode) throws Exception {
         VarHandle v = handle();
 
-        // Try a reflective invoke using a Method
+        // Try a reflective invoke using a Method, with no arguments
 
         Method vhm = VarHandle.class.getMethod(accessMode.methodName(), Object[].class);
         vhm.invoke(v, new Object[]{});
     }
 
+    @Test(dataProvider = "accessModesProvider")
+    public void methodInvocationMatchingArguments(VarHandle.AccessMode accessMode) throws Exception {
+        VarHandle v = handle();
+
+        // Try a reflective invoke using a Method, with an array of 0 arguments
+
+        Method vhm = VarHandle.class.getMethod(accessMode.methodName(), Object[].class);
+        Object args = new Object[0];
```

Comment:
I recommend naming this `arg`, as this is the single arg to the reflected method. Had you inlined this, you would have called `vhm.invoke(v, (Object) new Object[0]);`